feat: implement deployment of orogen-generated components in-process (on top of #413)

doudou opened the pull request:
Depends on:
- [ ] https://github.com/rock-core/tools-orogen_syskit_tests/pull/6
- [ ] https://github.com/rock-core/tools-orocosrb/pull/156
- [ ] https://github.com/rock-core/tools-orogen/pull/17
- [ ] https://github.com/rock-core/tools-syskit-log/pull/48

On top of #413

By "in-process", one means that the C++ component runs inside the Syskit process.

The functionality is accessed by declaring a deployment using `using_in_process_task TaskModel => "name"`

doudou committed:
feat: implement deployment of orogen-generated components in-process

By "in-process", one means that the C++ component runs inside the
Syskit process.

The functionality is accessed by declaring a deployment using
`using_in_process_task TaskModel => "name"`

doudou committed:
fix: create an in-process deployment model straight from the task

There is actually no value in looking for a task's default deployment
(plus, some tasks actually do not have one ... for instance logger::Logger)
Use Syskit's own deployment creation API to create the deployment model
straightaway

doudou committed:
feat: define Configuration#deregister_configured_deployment

Already exists on DeploymentGroup, but was not forwarded from Configuration

jhonasiv commented on lib/syskit/roby_app/in_process_tasks_manager.rb:
```diff
@@ -0,0 +1,163 @@
+            class NameMappingsForbidden < ArgumentError; end
+
+            # Fake status class
+            class Status
```
I dont see the point of this class, on wait_termination you mention returning a mapping with the status but in fact you are returning a mapping with the process. Also, isnt it possible to get the actual status with the in process task if its terminated?

doudou replied:
That's ... a good catch.

Essentially, wait_termination does not return any status. This actually causes every deployment using this process manager to emit failed on stop. The unmanaged task manager has the same issue.

doudou replied:
Fixed

doudou committed:
fix: return a proper status from InProcessTasksManager#wait_termination

The method was returning an array of processes, which was interpreted
by the runtime code as "stopped with error". Return a successful
status so that the deployment stops with success.

doudou committed:
fix: create the component loader before it is needed by the in-process task manager
